maplibre-gl.d.ts Syntax Error when using strict

I just started trying to use `maplibre-gl` today, and was following along with this conversation because I was unsuccessful in compiling types: https://github.com/maplibre/maplibre-gl-js/pull/787. After seeing this resolved and `v2.0.2` published, I tried it myself but still am getting errors.

**maplibre-gl-js version**: v2.0.2

**browser**: N/A

### Steps to Trigger Behavior

 1. Include and import `"maplibre-gl": "2.0.2"`
 2. Compile types

### Expected Behavior
Types compile

### Actual Behavior
There is a syntax error in the types output:
```
import type Pbf from 'pbf';
```
leading to the following errors:
```
../../node_modules/maplibre-gl/dist/maplibre-gl.d.ts:1:17 - error TS1005: '=' expected.

1     import type Pbf from 'pbf';
                  ~~~

../../node_modules/maplibre-gl/dist/maplibre-gl.d.ts:1:21 - error TS1005: ';' expected.

1     import type Pbf from 'pbf';
                      ~~~~

../../node_modules/maplibre-gl/dist/maplibre-gl.d.ts:1:26 - error TS1005: ';' expected.

1     import type Pbf from 'pbf';
                           ~~~~~
```

Thanks for reporting. My [debugging test repo](https://github.com/wipfli/maplibre-typescript/runs/4859914957?check_suite_focus=true) shows the same error with v2.0.2:

```
Error: node_modules/maplibre-gl/dist/maplibre-gl.d.ts(1,17): error TS1259: Module '"/home/runner/work/maplibre-typescript/maplibre-typescript/node_modules/@types/pbf/index"' can only be default-imported using the 'esModuleInterop' flag
Error: node_modules/maplibre-gl/dist/maplibre-gl.d.ts(20,53): error TS2503: Cannot find namespace 'GeoJSON'.
```

@HarelM  I have errors in angular also. Typescript 4.4

Error: node_modules/maplibre-gl/dist/maplibre-gl.d.ts:1:17 - error TS1259: Module '"/home/hex/xxx/ClientApp/node_modules/@types/pbf/index"' can only be default-imported using the 'allowSyntheticDefaultImports' flag

1     import type Pbf from 'pbf';
                  ~~~

  node_modules/@types/pbf/index.d.ts:82:1
    82 export = Pbf;
       ~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'allowSyntheticDefaultImports' flag.

I have tried to use the v2.0.2 dts file in the map-gl-indoor project (https://github.com/map-gl-indoor/map-gl-indoor/tree/feat/maplibre) but also encountered three TS errors.
```shell
> dts-bundle-generator -o dist/map-gl-indoor.d.ts src/index.ts
Compiling input files...
Processing src/index.ts
Writing src/index.ts -> dist/map-gl-indoor.d.ts
Checking generated files...
Compiler option "skipLibCheck" is disabled to properly check generated output
node_modules/maplibre-gl/dist/maplibre-gl.d.ts(2430,61): error TS2344: Type 'Us' does not satisfy the constraint '{}'.
  Type 'unknown' is not assignable to type '{}'.
node_modules/maplibre-gl/dist/maplibre-gl.d.ts(8797,4): error TS2411: Property 'mousemove' of type 'undefined' is not assignable to 'string' index type '(e: any) => void'.
node_modules/maplibre-gl/dist/maplibre-gl.d.ts(8798,4): error TS2411: Property 'mouseout' of type 'undefined' is not assignable to 'string' index type '(e: any) => void'.

Error: Compiled with errors
```
Unfortunately, `skipLibCheck` cannot be used as a workaround here.
**typescript version:** 4.5.2

Let me know if you prefer I open another issue
Similar errors found here: #474

```
Error: node_modules/maplibre-gl/dist/maplibre-gl.d.ts(3,8): error TS1259: Module '"/home/runner/work/maplibre-typescript/maplibre-typescript/node_modules/@types/mapbox__point-geometry/index"' can only be default-imported using the 'esModuleInterop' flag
```
was produced with v2.0.4 and a minimal TypeScript `app.ts` which reads:

```ts
import maplibregl from "maplibre-gl";
```

See [workflow run](https://github.com/wipfli/maplibre-typescript/runs/4913199673?check_suite_focus=true) in https://github.com/wipfli/maplibre-typescript

> You can check out if #800 solves this, not sure, I'm not entirely sure what the problem is. The definition file is correct as far as I know and understand.

It does not work with #800 :(

The #800 added me these errors:
```shell
node_modules/maplibre-gl/dist/maplibre-gl.d.ts(3,19): error TS7016: Could not find a declaration file for module '@mapbox/point-geometry'. '[...]/map-gl-indoor/node_modules/@mapbox/point-geometry/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/mapbox__point-geometry` if it exists or add a new declaration (.d.ts) file containing `declare module '@mapbox/point-geometry';`
node_modules/maplibre-gl/dist/maplibre-gl.d.ts(5,52): error TS7016: Could not find a declaration file for module '@mapbox/vector-tile'. '[...]/map-gl-indoor/node_modules/@mapbox/vector-tile/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/mapbox__vector-tile` if it exists or add a new declaration (.d.ts) file containing `declare module '@mapbox/vector-tile';`
```
that can be fixed following the tips, but the three previous remain.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@
 ### 🐞 Bug fixes
 
 - Add adjustment for glyph rendering, CJK fonts are mainly affected (#1002).
+- Improve typings to fix Angular strict mode failure (#790, #970, #934)
 - *...Add new stuff here...*
 
 ## 2.1.6

--- a/src/render/uniform_binding.ts
+++ b/src/render/uniform_binding.ts
@@ -7,7 +7,7 @@ type $ObjMap<T extends {}, F extends (v: any) => any> = {
     [K in keyof T]: F extends (v: T[K]) => infer R ? R : never;
 };
 
-export type UniformValues<Us extends any> = $ObjMap<Us, <V>(u: Uniform<V>) => V>;
+export type UniformValues<Us extends {}> = $ObjMap<Us, <V>(u: Uniform<V>) => V>;
 export type UniformLocations = {[_: string]: WebGLUniformLocation};
 
 abstract class Uniform<T> {

--- a/src/ui/map.ts
+++ b/src/ui/map.ts
@@ -917,7 +917,12 @@ class Map extends Camera {
         return this._rotating || this.handlers.isRotating();
     }
 
-    _createDelegatedListener(type: MapEvent, layerId: any, listener: any) {
+    _createDelegatedListener(type: MapEvent, layerId: string, listener: Listener):
+    {
+        layer: string;
+        listener: Listener;
+        delegates: {[type in keyof MapEventType]?: (e: any) => void};
+    } {
         if (type === 'mouseenter' || type === 'mouseover') {
             let mousein = false;
             const mousemove = (e) => {
@@ -1078,14 +1083,14 @@ class Map extends Camera {
             return super.on(type, layerIdOrListener as Listener);
         }
 
-        const delegatedListener = this._createDelegatedListener(type, layerIdOrListener, listener);
+        const delegatedListener = this._createDelegatedListener(type, layerIdOrListener as string, listener);
 
         this._delegatedListeners = this._delegatedListeners || {};
         this._delegatedListeners[type] = this._delegatedListeners[type] || [];
         this._delegatedListeners[type].push(delegatedListener);
 
         for (const event in delegatedListener.delegates) {
-            this.on(event as any, delegatedListener.delegates[event]);
+            this.on(event as MapEvent, delegatedListener.delegates[event]);
         }
 
         return this;
@@ -1133,10 +1138,10 @@ class Map extends Camera {
             return super.once(type, layerIdOrListener as Listener);
         }
 
-        const delegatedListener = this._createDelegatedListener(type, layerIdOrListener, listener);
+        const delegatedListener = this._createDelegatedListener(type, layerIdOrListener as string, listener);
 
         for (const event in delegatedListener.delegates) {
-            this.once(event as any, delegatedListener.delegates[event]);
+            this.once(event as MapEvent, delegatedListener.delegates[event]);
         }
 
         return this;

